src/components/RichTextEditor/Plugins/helper.tsx

```diff
@@ -25,3 +27,110 @@ export const ButtonGroupWrapper = ({ index, children }: ButtonGroupProps) => (
+            textTransform: `var(${PREFIX}${textStyle}-text-transform)` as CSSProperties['textTransform'],
+        };
+    }
+    return {
+        color: `var(${PREFIX}${textStyle}-color)`,
+        fontFamily: `var(${PREFIX}${textStyle}-font-family)`,
+        fontSize: `var(${PREFIX}${textStyle}-font-size)`,
+        fontStyle: `var(${PREFIX}${textStyle}-font-style)`,
+        fontWeight: `var(${PREFIX}${textStyle}-font-weight)`,
+        letterSpacing: `var(${PREFIX}${textStyle}-letter-spacing)`,
+        lineHeight: `var(${PREFIX}${textStyle}-line-height)`,
+        textDecoration: `var(${PREFIX}${textStyle}-text-decoration)`,
+        textTransform: `var(${PREFIX}${textStyle}-text-transform)` as CSSProperties['textTransform'],
+    };
+};
+
+export const getButtonStyleCssProperties = (
+    buttonStyle: RichTextButtonStyle,
+): CSSProperties & { hover: CSSProperties } => {
+    const styles = {
+        fontSize: `var(${PREFIX}button-${buttonStyle}-font-size)`,
+        color: `var(${PREFIX}button-${buttonStyle}-color)`,
+        paddingTop: `var(${PREFIX}button-${buttonStyle}-padding-top)`,
+        paddingRight: `var(${PREFIX}button-${buttonStyle}-padding-right)`,
+        paddingBottom: `var(${PREFIX}button-${buttonStyle}-padding-bottom)`,
+        paddingLeft: `var(${PREFIX}button-${buttonStyle}-padding-left)`,
+        fontFamily: `var(${PREFIX}button-${buttonStyle}-font-family)`,
+        fontStyle: `var(${PREFIX}button-${buttonStyle}-font-style)`,
+        fontWeight: `var(${PREFIX}button-${buttonStyle}-font-weight)`,
+        textTransform: `var(${PREFIX}button-${buttonStyle}-text-transform)` as CSSProperties['textTransform'],
+        lineHeight: `var(${PREFIX}button-${buttonStyle}-line-height)`,
+        borderColor: `var(${PREFIX}button-${buttonStyle}-border-color)`,
+        borderWidth: `var(${PREFIX}button-${buttonStyle}-border-width)`,
+        borderRadius: `var(${PREFIX}button-${buttonStyle}-border-radius)`,
+        backgroundColor: `var(${PREFIX}button-${buttonStyle}-background-color)`,
+        hover: {
+            color: `var(${PREFIX}button-${buttonStyle}-hover-color)`,
+            borderColor: `var(${PREFIX}button-${buttonStyle}-hover-border-color)`,
+            backgroundColor: `var(${PREFIX}button-${buttonStyle}-hover-background-color)`,
+        },
+    };
+    return { ...styles, cursor: 'pointer', display: 'inline-block', margin: '10px 0' };
+};
+
+export const getLiStyles = (element: TElement): CSSProperties => {
+    return {
+        ...getTextStyleCssProperties(getDeepestTextStyle(element)),
+        counterIncrement: 'count',
+    };
+};
+
+const getDeepestTextStyle = (node: TElement): string => {
+    let deepestTextStyle;
+
+    if (node.type === 'a') {
+        deepestTextStyle = node.children[0].textStyle;
+    } else if (node.children) {
+        for (const childNode of node.children) {
+            const childDeepestTextStyle = getDeepestTextStyle(childNode as TElement);
```

Comment:
Imho it is already clear that it is a child and you could rename it to `deepestTextStyle` and change let to `let textStyle`. Maybe this makes it a bit easier to read. wdyt?